addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -647,74 +659,63 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
-                    });
+                    genre = messagePart.substring("genre%3A".length());
                 }
                 // Parameter Year
                 else if (messagePart.startsWith("year%3A")) {
-                    final String value = messagePart.substring("year%3A".length());
-                    updatePlayer(new PlayerUpdateEvent() {
-                        @Override
-                        public void updateListener(SqueezeBoxPlayerEventListener listener) {
-                            listener.yearChangeEvent(mac, decode(value));
-                        }
-                    });
+                    year = messagePart.substring("year%3A".length());
                 }
-                // Parameter Artwork
-                else if (messagePart.startsWith("artwork_track_id%3A")) {
-                    String url = messagePart.substring("artwork_track_id%3A".length());
-                    // NOTE: what is returned if not an artwork id? i.e. if a
-                    // space?
-                    if (!url.startsWith(" ")) {
-                        url = "http://" + host + ":" + webport + "/music/" + url + "/cover.jpg";
+                // Parameter Artwork_url
+                else if (messagePart.startsWith("artwork_url%3A")
+                        && !messagePart.startsWith("artwork_url%3Ahttp%3A%2F%2F")) {
+                    url = messagePart.substring("artwork_url%3A".length());
+                    if (url.startsWith("%2F")) {
+                        url = "http://" + host + ":" + webport + decode(url);
+                    } else {
+                        url = "http://" + host + ":" + webport + "/" + decode(url);
                     }
-                    final String value = url;
-                    updatePlayer(new PlayerUpdateEvent() {
-                        @Override
-                        public void updateListener(SqueezeBoxPlayerEventListener listener) {
-                            listener.coverArtChangeEvent(mac, decode(value));
-                        }
-                    });
                 } else {
                     // Added to be able to see additional status message types
                     logger.trace("Unhandled status message type '{}'", messagePart);
                 }
             }
+            // update cover art
+            // update remote title
+            // update artist
+            // update album
+            // update genre
+            // update year
+
+            final String f_url = url;
+            final String f_remoteTitle = remoteTitle;
+            final String f_artist = artist;
+            final String f_album = album;
+            final String f_genre = genre;
+            final String f_year = year;
+
+            updatePlayer(new PlayerUpdateEvent() {
+                @Override
+                public void updateListener(SqueezeBoxPlayerEventListener listener) {
+                    listener.coverArtChangeEvent(mac, f_url);
```

Comment:
Are you now overwriting the values with null or is there some other mechanism in place to guard that?

Reply:
Or is everything always present in the message?

Reply:
This values are always present in the message (in the same song). If not (when you change music to web radio for example), values ares initialised with empty string (line 546)

Reply:
Is an empty string proper or would null be way more proper to indicate that no song is playing

Reply:
Sorry i not explained properly.
Not all value are present in the message. It depend what you are playing. For example if you playing a web radio  "rtitle" is present, but if you play local music, "rtitle" is not present in the message. So a empty string is not always indicate that no song is playing